packages/contract/lib/handlers.js

```diff
@@ -101,6 +101,9 @@ const handlers = {
       return context.promiEvent.reject(error);
     }
 
+    // Check logs for Console.log usage & log if necessary
+    Utils.consoleLog(receipt.logs);
```

Comment:
There's some nasty business with `receipt.logs` and `receipt.rawLogs`, and `receipt.logs` changes during the lifecycle of the transaction. `rawLogs` is straight from Web3.js, I believe, and might be preferable to use, unless you've done your due diligence here to make sure this doesn't have weird edge cases.

I would think to check Ganache in standard mode vs. default mode (`--noVMErrorsOnRPCResponse` flag), and also check for constructor behavior vs. regular contract call behavior. (Constructors can emit logs, right?)

cc @haltman-at because you know the perils of `receipt.logs`, if you have any insight here

Reply:
For controlled logs like this, I think using `logs` should probably be fine.  Using `rawLogs` would mean having to do the decoding rather than getting them pre-decoded.  So, I think using `logs` here is fine.  As we know, the existing decoding logic that's used for `logs` (i.e.: ethers) has a number of problems, but if they used `rawLogs` they'd either just be:

1. Invoking that same decoder themselves, which wouldn't improve matters any;
2. Doing things manually, which doesn't strike me as a great idea; or
3. Using Truffle Decoder, which I don't think it currently makes sense to use here.

So, I think this way of doing things is basically fine for now.

Reply:
Thanks @haltman-at for the input, will resolve this for now.